File: csharp/Behavioral/15-Iterator/Iterator/PriorityJobIterator.cs
public class PriorityJobIterator : IJobIterator
{
    private readonly List<JobCard> _sorted;
    private int                    _index = 0;

    public PriorityJobIterator(IEnumerable<JobCard> jobs)
    {
        _sorted = jobs
            .OrderByDescending(j => j.Priority)
            .ToList();
    }

    public bool    HasNext() => _index < _sorted.Count;
    public JobCard Next()    => _sorted[_index++];
    public void    Reset()   => _index = 0;
}

File: typescript/Behavioral/15-Iterator/Iterator/PriorityJobIterator.ts
import { JobCard } from "../Model/JobCard";

export class PriorityJobIterator {
  private sorted: JobCard[];
  private index = 0;

  constructor(jobs: JobCard[]) {
    this.sorted = [...jobs].sort((a, b) => b.priority - a.priority);
  }

  hasNext(): boolean {
    return this.index < this.sorted.length;
  }
  next(): JobCard {
    return this.sorted[this.index++];
  }
  reset(): void {
    this.index = 0;
  }
}
